ariadne/asgi.py

```diff
@@ -237,10 +249,16 @@ async def handle_websocket_message(
         message_type = cast(str, message.get("type"))
 
         if message_type == GQL_CONNECTION_INIT:
+            if callable(self.on_connect):
+                await self.on_connect(message, websocket)
```

Comment:
How can I make the authentication state I've resolved within `OnConnect` accessible in my subscription sources and resolvers? And what if `OnConnect` and `OnDisconnect` aren't async?

Shouldn't we also make context available to those functions in the case when it contains data that the auth process depends on? 🤔

Reply:
Yeah, we could modify it to handle non-async functions, and also pass in the context. I made this as a minimum change to get my software working and communicate a solution.

> How can I make the authentication state I've resolved within OnConnect accessible in my subscription sources and resolvers?

One approach could be to set an `auth_token` attribute on the websocket object within `on_connect`, then authentication can be carried out in the `context_value` function where that websocket is passed. There could be other solutions like using `Sec-WebSocket-Key` header as an ID and associating that with the `auth_token` or `user` in Redis.